resources/js/photoinit.js

```diff
@@ -178,12 +175,15 @@ function initPhotoSwipeFromDOM (gallerySelector) {
     $('.pswp__button--mail').on('click touchstart', function (e) {
         e.preventDefault();
         e.stopPropagation();
-
-        const img = gallery.currItem.src.split('/').pop();
-
         photoBooth.toggleMailDialog(img);
     });
 
+    $('.pswp__button--download').on('click touchstart', function (e) {
+        e.preventDefault();
+        e.stopPropagation();	
+        location = 'api/download.php?image=' + img;

```

Comment:
`window.location` would be better than `location` (because it makes clear that `location` is no local variable), but the [html5 download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) attribute would be even better. If you create a new `a` element at this position with the corresponding attributes a emulated click event on that element should work.

Reply:
If I get you right, you suggest to replace `<button class="pswp__button pswp__button--download" ...` in `pwswp.template.php` (i.e. the current approach that applies to all the other buttons there) with an `<a href="" download="<js-injected-url>" ...` ?

Reply:
No, I meant something like this (untested):

```
const linkElement = $('<a>');
linkElement.attr({
   href: 'URL_TO_IMAGE',
   download: 'FILENAME',
});
linkElement.click();
```

This should ensure that the browser shows the download prompt instead of open the image.

Reply:
Also check if download attribute is supported by the browser?
```
function doesAnchorSupportDownload() {
    return "download" in document.createElement("a")
}
```
Currently I am checking it this way
```
        if (doesAnchorSupportDownload()) {
            location = 'api/download.php?image=' + img;
        } else {
            alert('Unsupported on your Browser');
        }
```

Reply:
Sorry, I’m afraid I don’t get it. „On click“ the element gets manipulated and then .click() gets called? What’s the benefit?

I guess I will leave this PR open and leave it up to you (or s.o. else with web developer skills) to properly implement this feature. 😌

Reply:
@sualko Out of curiosity I still tried to fix this according to your suggestions, see: https://github.com/andreknieriem/photobooth/pull/183/commits/7d986471fd54b9f9b1e5f1f0f49c0102f64a651c . It works but might be complete bullsh... Looking forward to your feedback. ;-)